lib/metric-config-parser/metric_config_parser/data_source.py

```diff
@@ -90,6 +101,7 @@ class DataSource:
     description = attr.ib(default=None, type=str)
     joins = attr.ib(default=None, type=List[DataSourceJoin])
     columns_as_dimensions = attr.ib(default=False, type=bool)
+    aggregation_units = attr.ib(default=[AnalysisUnit.CLIENT], type=List[AnalysisUnit])
```

Comment:
Without a validator, is this guaranteed to be of type list? The doc string says its optional, but are data sources ever initialized with `aggregation_units = None`? If that's the case, at runtime the type of this parameter could be `None` despite the type annotation saying `list[AnalysisUnit]`.

Reply:
Hm.. I think that should be disallowed. And I don't think you can pass the literal python `None` as a value, it would have to be a string, which would raise a `ValueError` the same way any other string does.

Reply:
Right, toml doesn't support null/None. If this field wouldn't have a default value and would not be explicitly set, then that's when it would be `None`